packages/multichain-account-service/src/providers/BaseBip44AccountProvider.ts

```diff
@@ -80,7 +101,7 @@ export abstract class BaseBip44AccountProvider<
    */
   init(accounts: Account['id'][]): void {
     for (const account of accounts) {
-      this.accountsList.add(account);
+      this.accounts.add(account);
```

Comment:
WDYT about adding a small comment saying we assume the caller is always sending valid/compatible accounts for this provider?

Here, nothing prevents me from doing the wrong accounts (which could mess up the entire logic).

So either we `assert(this.isAccountCompatible(account), 'Incompatible account')` here (should be a pretty cheap call I think)

OR we assume the caller uses the right accounts right away.

If we go without the `assert` I'd go add this comment/note in the jsdoc directly

Reply:
I'll add the comment! However the init call is piped from the service's filtered accounts so I don't think it would ever be an issue. Do we see the provider classes being used outside of the context of the service?